### PR TITLE
Fix AddEventLog: EventLogSettings not added 

### LINKED ISSUE #51899: AddEventLog settings silently disappear
### Description

If you call `EventLoggerFactoryExtensions.AddEventLog(this ILoggingBuilder builder, EventLogSettings settings)` on a service collection that already contains Event Log support (e.g. one created using `Host.CreateDefaultBuilder()`), the provided settings are silently ignored. This is because that method uses `TryAddEnumerable()`, which does nothing if the service is already present.

### Configuration

.NET 5.0 (latest as of report date)
Windows 10 x64
Specific to Windows (Event Log is Windows-only)

### Regression?

This worked in .NET Core 2.1 because the default builder did not add Event Log support by default, thus `TryAddEnumerable()` did in fact add the new logging provider with the specified settings. I'm not sure about the .NET Core 3.1 behavior; we are skipping past it.

### Other information

This could be treated as a documentation error; `EventLoggerFactoryExtensions.AddEventLog(this ILoggingBuilder builder, Action<EventLogSettings> configure)` does not have the issue because it always registers the provided callback. [The ASP.NET Core documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0#windows-eventlog) and/or the method comments could be updated to indicate this behavior clearly.

### PR DESCRIPTION
**Pull request by wrzucher**
TryAddEnumerable do not add second provider to service collection

Closes #51899

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=57567) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=57567) before we can accept your contribution.<br/><br/>:x: wrzucher [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=57567)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=57567) it.</sub>

File: src/libraries/Microsoft.Extensions.Logging.EventLog/src/EventLoggerFactoryExtensions.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public static ILoggingBuilder AddEventLog(this ILoggingBuilder builder, EventLog
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<ILoggerProvider>(new EventLogLoggerProvider(settings)));
```
[the remark this answers is not in the document]
Comment (maryamariyan) [reply src/libraries/Microsoft.Extensions.Logging.EventLog/src/EventLoggerFactoryExtensions.cs:51]: I agree, I prefer the other two overloads to `AddEventLog(this ILoggingBuilder builder, EventLogSettings settings)`. If we wanted to add event log with settings we should be using the overload taking `Action<EventLogSettings>`.

Looked at the PR again I think it looks good as-is. Just added a nit review comment.

Thanks

File: src/libraries/Microsoft.Extensions.Logging/tests/Common/EventLogLoggerTest.cs [hunk 2]
```diff
@@ -126,6 +126,34 @@ public void IOptions_CreatesWindowsEventLog_WithSuppliedEventLogSettings()
...
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+            var eventLogSettings = new EventLogSettings()
+            {
+                SourceName = "foo",
+                LogName = "bar",
+                MachineName = "blah",
+                EventLog = null,
+            };
+            serviceCollection.AddLogging(builder => builder.AddEventLog());
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Logging/tests/Common/EventLogLoggerTest.cs:142]: This line can be removed

**Review by maryamariyan — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Logging/tests/Common/EventLogLoggerTest.cs [hunk 3]
```diff
@@ -126,6 +126,34 @@ public void IOptions_CreatesWindowsEventLog_WithSuppliedEventLogSettings()
...
+
+            // Act
+            serviceCollection.AddLogging(builder => builder.AddEventLog(eventLogSettings));
+
+            // Assert
+            var services = serviceCollection.BuildServiceProvider();
```
Comment (deeprobin) [inline src/libraries/Microsoft.Extensions.Logging/tests/Common/EventLogLoggerTest.cs:148]: Usage explicit type
```suggestion
            IServiceProvider services = serviceCollection.BuildServiceProvider();
```

**Review by deeprobin — CHANGES_REQUESTED** [review]

**Review by deeprobin — APPROVED** [review]
LGTM. If CI passes, I think its ready to merge.

**Comment by wrzucher** [comment]
@maryamariyan please, confirm that you take a look at my concerns about this PR in the original issue
https://github.com/dotnet/runtime/issues/51899#issuecomment-900804993

**Comment by maryamariyan** [comment]
Thanks for the note @wrzucher. I also believe the proper overload to use in general is the one below:

```c#
public static ILoggingBuilder AddEventLog(this ILoggingBuilder builder, Action<EventLogSettings> configure)
```

And I think the usage to the overload being fixed in this PR should be discouraged. (Note that `AddConsole` for example does not have such an overload, we take `Action` instead.)
I think it's better to close this PR rather than merge because there is not much value-add in fixing this API overload. Also since the current fix is not without risks.

**Pull request closed without merging**